redaxo/src/addons/install/lib/archive.php

```diff
@@ -10,22 +10,32 @@ class rex_install_archive
     public static function extract($archive, $dir, $basename = '')
     {
         $dir = rtrim($dir, '/\\');
+        rex_dir::delete($dir);
```

Comment:
wenn man in ein verzeichnis rein-extrahiert wo schon dateien drinn sind, würden die jetzt gelöscht werden... vor der Änderung hatten diese überlebt.
(also auch dateien die überhaupt nix mit dem archiv zu tun hatten)

Reply:
Ist das nicht sogar erwünschtes Verhalten?

Reply:
In einer public-Api würde ich es anders machen, aber die Klasse hier ist internal im Installer, und da ist an allen Nutzungsstellen das Verhalten gewünscht. Bzw. diese Zeile löst genau das Hauptproblem, worüber einige gestolpert sind.

Ich kann es auch an alle Nutzungsstellen packen (vor Aufruf von `extract()`). Aber wie gesagt, da das hier internal ist, denke ich, ist es auch so ok, oder?

Reply:
ok, da die klasse nur für achive im installer-context ist - passt es so. 👍